pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -232,6 +233,17 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	return setStatus(u, status)
 }
 
+// aws.accessRole.Identity is a output-only field in CREATE, it is required in the UPDATE.
+func merge3(desired *pb.Connection, previouslyApplied *krm.BigQueryConnectionConnectionObservedState) *pb.Connection {
+	if previouslyApplied != nil {
+		if previouslyApplied.Aws != nil {
```

Comment:
nit: `if previouslyApplied != nil && previouslyApplied.Aws != nil {`

Reply:
Sure, done.
(I was thinking we may want to deal with other property merge, like Azure, SQL, Spanner, Spark, Salesforce)